tests/test_functions.py

```diff
@@ -455,3 +456,32 @@ def test_class_in_main_is_in_right_module():
     resp = client.get("/")
 
     assert resp.status_code == 200
+
+
+def test_legacy_function_check_env():
+    source = TEST_FUNCTIONS_DIR / "http_check_env" / "main.py"
+    target = "function"
+
+    os.environ["ENTRY_POINT"] = target
+
+    client = create_app(target, source).test_client()
+    resp = client.post("/", json={"mode": "FUNCTION_TRIGGER_TYPE"})
+    assert resp.status_code == 200
+    assert resp.data == b"http"
+
+    resp = client.post("/", json={"mode": "FUNCTION_NAME"})
+    assert resp.status_code == 200
+    assert resp.data == b"function"
+
+
+def test_legacy_function_returns_none():
```

Comment:
Can you also test that non-legacy functions handle None correctly?

Reply:
Added this test